pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/OptionsTest.java

```diff
@@ -417,6 +423,34 @@ public void withCredentialsStageWrapper() throws Exception {
                 .go();
     }
 
+    @Issue("JENKINS-50561")
+    @Test
+    public void rateLimitBuilds() throws Exception {
+        WorkflowRun b = expect("rateLimitBuilds")
+                .go();
+        WorkflowJob p = b.getParent();
+
+        RateLimitBranchProperty.JobPropertyImpl prop = p.getProperty(RateLimitBranchProperty.JobPropertyImpl.class);
+        assertNotNull(prop);
+        assertEquals(1, prop.getCount());
+        assertEquals("day", prop.getDurationName());
+        assertFalse(prop.isUserBoost());
+
+        QueueTaskFuture<WorkflowRun> inQueue = p.scheduleBuild2(0);
+
+        while (!Queue.getInstance().contains(p)) {
+            Thread.yield();
```

Comment:
Is there any particular reason you're using `Thread#yield` instead of `Thread#sleep`?

Reply:
Cargo-culted from [here](https://github.com/jenkinsci/branch-api-plugin/blob/master/src/test/java/jenkins/branch/RateLimitBranchPropertyTest.java#L167), so no, no particular reason.